cfile/cfile.cpp

```diff
@@ -103,42 +94,38 @@ static CFILE *open_file_in_lib(const char *filename);
 // NOTE:	libname must be valid for the entire execution of the program.  Therefore, it should either
 //			be a fully-specified path name, or the current directory must not change.
 // Returns: 0 if error, else library handle that can be used to close the library
-int cf_OpenLibrary(const char *libname) {
+int cf_OpenLibrary(const std::filesystem::path &libname) {
   FILE *fp;
   int i;
   uint32_t offset;
   static int first_time = 1;
-  tHogHeader header;
-  tHogFileEntry entry;
+  tHogHeader header{};
+  tHogFileEntry entry{};
 
   // allocation library structure
   std::shared_ptr<library> lib = std::make_shared<library>();
 #ifdef __LINUX__
   // resolve library name
-  char t_dir[_MAX_PATH];
-  char t_file[_MAX_PATH];
-  char t_ext[256];
-  ddio_SplitPath(libname, t_dir, t_file, t_ext);
-  const char *resolve_dir = nullptr;
-  const char *resolve_name = libname;
-  if (strlen(t_dir) > 0) {
-    strcat(t_file, t_ext);
-    resolve_dir = t_dir;
-    resolve_name = t_file;
+  std::filesystem::path resolve_dir = libname.parent_path();
+  std::filesystem::path resolve_name = libname;
+
+  if (!resolve_dir.empty()) {
+    resolve_name = libname.filename();
   }
-  char t_out[_MAX_PATH];
-  if (!cf_FindRealFileNameCaseInsenstive(resolve_dir, resolve_name, t_out)) {
+
+  std::filesystem::path t_out = cf_FindRealFileNameCaseInsensitive(resolve_name, resolve_dir);
+  if (t_out.empty()) {
     return 0; // CF_NO_FILE
   }
   // re-assemble
-  if (resolve_dir != nullptr)
-    ddio_MakePath(lib->name, resolve_dir, t_out, nullptr);
+  if (!resolve_dir.empty())
+    lib->name = resolve_dir / t_out;
   else
-    strcpy(lib->name, t_out);
+    lib->name = t_out;
 #else
-  strcpy(lib->name, libname);
+  lib->name = libname;
```

Comment:
we could have the same code path for case sensitive and case insensitive fs, if`cf_FindRealFileNameCaseInsensitive` works for both. It does not have a big performance impact